widget/label.go

```diff
@@ -45,6 +42,7 @@ func NewLabelWithStyle(text string, alignment fyne.TextAlign, style fyne.TextSty
 		TextStyle: style,
 	}
 
+	l.ExtendBaseWidget(l.super())
```

Comment:
Why is the `l.super()` needed now? 🤔

Reply:
It's a good question but the extend tests fail if you use `l.ExtendBaseWidget(l)` instead

Reply:
The answer is probably in how the Label's replacement ExtendBaseWidget works

Reply:
I see, however it seems that the right thing to do is using it inside ExtendBaseWidget reimplementation. By doing this:
```go
func (l *Label) ExtendBaseWidget(w fyne.Widget) {
	if w == nil {
		w = l
	}
	l.BaseWidget.ExtendBaseWidget(w)
	if l.provider != nil {
		l.provider.ExtendBaseWidget(l.super()) // 👈
	}
}
```
In that way we could remove all the `l.ExtendBaseWidget(l.super())` :)
```diff
diff --git a/widget/label.go b/widget/label.go
index b0d70842..d93c484a 100644
--- a/widget/label.go
+++ b/widget/label.go
@@ -42,7 +42,7 @@ func NewLabelWithStyle(text string, alignment fyne.TextAlign, style fyne.TextSty
                TextStyle: style,
        }
 
-       l.ExtendBaseWidget(l.super())
+       l.ExtendBaseWidget(l)
        return l
 }
 
@@ -60,7 +60,7 @@ func (l *Label) Bind(data binding.String) {
 // CreateRenderer is a private method to Fyne which links this widget to its renderer
 func (l *Label) CreateRenderer() fyne.WidgetRenderer {
        l.provider = NewRichTextWithText(l.Text)
-       l.ExtendBaseWidget(l.super())
+       l.ExtendBaseWidget(l)
        l.syncSegments()
 
        return l.provider.CreateRenderer()
@@ -73,7 +73,7 @@ func (l *Label) ExtendBaseWidget(w fyne.Widget) {
        }
        l.BaseWidget.ExtendBaseWidget(w)
        if l.provider != nil {
-               l.provider.ExtendBaseWidget(w)
+               l.provider.ExtendBaseWidget(l.super())
        }
 }
```